tests/kernel/wave/speculative_decode_test.py

```diff
@@ -109,32 +116,35 @@ def tree_speculative_sampling_target_only(
                 break
 
         accept_token_num[bx] = num_accepted_tokens
+        cur_prob_offset_vec[bx] = cur_prob_offset
+        last_accepted_retrive_idx_vec[bx] = last_accepted_retrive_idx
 
-        # Sample from relu(target_probs - draft_probs)
-        last_offset = cur_prob_offset
-        q = target_probs[bx, last_offset]
-        p = (
-            draft_probs[bx, last_offset]
-            if num_accepted_tokens != num_speculative_tokens - 1
-            else torch.zeros_like(q)
-        )
-        relu_diff = torch.zeros_like(q)
-        wave_kernel(q, p, relu_diff)
-        sum_relu = relu_diff.sum()
+    # Sample from relu(target_probs - draft_probs)
+    relu_diff = torch.zeros_like(target_probs)
+    u = torch.zeros(
+        [target_probs.shape[0], target_probs.shape[1]], device=target_probs.device
+    )
+    wave_kernel(
+        target_probs,
+        draft_probs,
+        cur_prob_offset_vec,
+        uniform_samples[:, 0],
```

Comment:
why do we only pass a slice of the uniform samples?

Reply:
I guess the refrence only takes 0th index, but I can move that to kernel/mapping as well

Reply:
done!